ignite/pkg/jsonfile/jsonfile.go

```diff
@@ -0,0 +1,311 @@
+package jsonfile
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/pkg/errors"
+
+	"github.com/ignite-hq/cli/ignite/pkg/tarball"
+)
+
+const (
+	keySeparator = "."
+)
+
+type (
+	// JSONFile represents the file
+	JSONFile struct {
+		file        readWriteSeeker
+		tarballPath string
+		url         string
+		updates     map[string][]byte
+		cache       []byte
+	}
+	// UpdateFileOption configures file update.
+	UpdateFileOption func(map[string][]byte)
+
+	writeTruncate interface {
+		Truncate(size int64) error
+	}
+	readWriteSeeker interface {
+		io.ReadWriteSeeker
+		Close() error
+		Sync() error
+	}
+)
+
+var (
+	// ErrFieldNotFound parameter not found into json
+	ErrFieldNotFound = errors.New("JSON field not found")
+	// ErrInvalidValueType invalid value type
+	ErrInvalidValueType = errors.New("invalid value type")
+	// ErrInvalidURL invalid file URL
+	ErrInvalidURL = errors.New("invalid file URL")
+)
+
+// New creates a new JSONFile
+func New(file readWriteSeeker) *JSONFile {
+	return &JSONFile{
+		updates: make(map[string][]byte),
+		file:    file,
+	}
+}
+
+// FromPath parse JSONFile object from path
+func FromPath(path string) (*JSONFile, error) {
+	file, err := os.OpenFile(path, os.O_RDWR|os.O_APPEND, 0600)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot open the file")
+	}
+	return New(file), nil
+}
+
+// FromURL fetches the file from the given URL and returns its content.
+func FromURL(ctx context.Context, url, path, tarballFileName string) (*JSONFile, error) {
+	// TODO create a cache system to avoid download genesis with the same hash again
```

Comment:
We can cache the genesis files under `~/.ignite/genesises` by giving random/unique names to each and we can use https://github.com/ignite-hq/cli/blob/develop/ignite/pkg/cache to keep genesis path/hash pair.

Reply:
Great, let's keep this PR because it's already big, and we can implement the cache in another one. I created an issue for it.

https://github.com/ignite-hq/cli/issues/2550